packages/server/src/arpa_reporter/lib/audit-report.js

```diff
@@ -139,47 +141,93 @@ async function createObligationSheet(periodId, domain, tenantId, logger = log) {
 
 async function createProjectSummaries(periodId, domain, tenantId, logger = log) {
     logger.info('building rows for spreadsheet');
-    const records = await mostRecentProjectRecords(periodId, tenantId);
-    logger.fields.sheet.totalRecentRecords = records.length;
-    logger.info('retrieved most recent project records');
-
-    const rows = await Promise.all(records.map(async (record) => {
-        const recordLogger = logger.child({
-            record: {
-                upload: {
-                    id: record.upload.id,
-                    reportingPeriod: { id: record.upload.reporting_period_id },
-                },
-                project: {
-                    id: record.content.Project_Identification_Number__c,
-                },
-            },
-        });
-        recordLogger.debug('populating row from record');
-        const reportingPeriod = await getReportingPeriod(
-            record.upload.reporting_period_id, undefined, tenantId,
+    const uploads = await knex('uploads')
```

Comment:
Can we move this to a function inside of `db`? Not sure if we will use it anywhere else, but it does help keep the functions in here smaller.

Reply:
For the moment, my intent was to explicitly avoid query reuse (at the expense of larger LoC in the function) until we can see how applicable this query pattern is for the other sheets in the audit report workbook (or similar workflows).

I have no issues ultimately moving/repurposing/refactoring this if it ends up being more broadly usable, but one of the issues I'd found was contributing to technical debt around this problem was the prevalence of query functions being reused and expanded over time (as opposed to, say, writing a more performant, but single-use query) and I'd like to avoid that until we have a more holistic strategy for getting what we need from the DB in these workflows.

Reply:
Makes sense. Thank you for the explanation.